src/python/pants/backend/terraform/goals/lockfiles.py

```diff
@@ -109,8 +117,31 @@ async def generate_lockfile_from_sources(
         ),
     )
 
+    args = ["providers", "lock"]
+    args.extend(terraform_arg("-platform", platform) for platform in lockfile_request.platforms)
```

Comment:
Nit: This could be folded into the prior line by adding `*(terraform_arg("-platform", platform) for platform in lockfile_request.platforms)` to the list.